From 1bd606110942d5014c9eee967c7222862936a20d Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Thu, 23 Jul 2026 15:44:13 -0700
Subject: [PATCH 13105/15333] apparmor: fix error debug output in
 fn_label_build

checking PTR_ERROR() is not correct to just determine if any error
occured, instead use the IS_ERR macro and also output the PTR_ERR
as part of the debug message.

Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 security/apparmor/include/lib.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/security/apparmor/include/lib.h b/security/apparmor/include/lib.h
index e3c8cb044a90..d59eee5603a4 100644
--- a/security/apparmor/include/lib.h
+++ b/security/apparmor/include/lib.h
@@ -338,8 +338,8 @@ __do_cleanup:								\
 		__new_ = (FN);						\
 	}								\
 __done:									\
-	if (PTR_ERR(__new_))						\
-		AA_DEBUG(DEBUG_LABEL, "label build failed\n");		\
+	if (IS_ERR(__new_))						\
+		AA_DEBUG(DEBUG_LABEL, "label build failed %ld\n", PTR_ERR(__new_)); \
 	(__new_);							\
 })
 
-- 
2.55.0

